app/models/solidus_bolt/gateway.rb

```diff
@@ -45,8 +45,29 @@ def capture(float_amount, response_code, gateway_options)
       ActiveMerchant::Billing::Response.new(false, e, {})
     end
 
-    def void(_response_code, _gateway_options)
-      raise NotImplementedError, 'void method has not been implemented in SolidusBolt::Gateway class'
+    def void(response_code, gateway_options)
+      payment = gateway_options[:originator]
+      payment_source = payment.source
+      payment_method = payment.payment_method
+
+      raise 'Payment is not in the correct state to be voided' unless can_void?(payment)
```

Comment:
It looks to me that `void` methods on other extensions don't make this check, probably because it's already executed under `can_void?` condition.
Can we remove this check?

Reply:
removed